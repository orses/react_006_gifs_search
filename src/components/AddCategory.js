import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = e => {
    setInputValue(e.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    // updating categories state
    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="search"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Type a category and hit Enter"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
