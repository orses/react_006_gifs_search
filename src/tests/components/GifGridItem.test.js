import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe("Test over GifGridItem component", () => {
  const title = "Elemento individual a renderizar";
  const port = 3000;
  const url = process.env.HOST_LOCAL || `http://localhost:${port}`;
  const id = "12345";
  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a paragraph with the title text", () => {
    const textRendered = wrapper.find("p").text();
    expect(textRendered).toBe(title);
  });

  test("should to have an image with url and alt", () => {
    const imgRendered = wrapper.find("img");
    expect(imgRendered.prop("src")).toBe(url);
    expect(imgRendered.prop("alt")).toBe(title);
  });

  test("should has class animate__fadeIn", () => {
    const className = "animate__fadeIn";
    const cssRendered = wrapper.find("div");

    expect(cssRendered.prop("className")).toContain(className);
  });
});
