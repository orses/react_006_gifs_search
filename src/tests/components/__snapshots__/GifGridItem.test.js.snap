// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test over GifGridItem component should render the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Elemento individual a renderizar"
    src="http://localhost:3000"
  />
  <p>
    Elemento individual a renderizar
  </p>
</div>
`;
